docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,766 @@
+
+`info`:  a list of `PushNotificationQueryInfo`
+
+### PushNotification
+
+`access_token`: the access token used to send a push notification
+`chat_id`: the `SHA3-256` of the `chat_id`
+
+### Data disclosed
+
+- The `SHA3-256` of the `chat_id` the notification is to be sent for
+
+### PushNotificationRequest
+
+`requests`: a list of `PushNotification`
+`message`: the encrypted message that we want to notify on
+`message_id`: the [status message id](./6-payloads.md)
+`ack_required`: a 32 bytes long AES key, only set if an ack is required
+
+### Data disclosed
+
+- The status message id for which the notification is for
+- the cypher text of the message
+
+### PushNotificationAcknowledgement
+
+`id`: the `ack_required` string passed by the client
+
+## Partitioned topic
+
+This is a modification of the [partitioned topic](./10-waku-usage.md) used by clients
+
+```golang
+var partitionsNum *big.Int = big.NewInt(5000)
+var partition *big.Int = big.NewInt(0).Mod(publicKey.X, partitionsNum)
+
+partitionTopic := "push-notifications-" + strconv.FormatInt(partition.Int64(), 10)
+
+var hash []byte = keccak256(partitionTopic)
+var topicLen int = 4
+
+if len(hash) < topicLen {
+    topicLen = len(hash)
+}
+
+var topic [4]byte
+for i = 0; i < topicLen; i++ {
+    topic[i] = hash[i]
+}
+```
+
+
+## Anonymous mode of operations
+
+An anonymous mode of operations MAY be provided by the client, where the 
+responsibility of propagating information about the user is left to the client,
+in order to preserve privacy.
+
+A client in anonymous mode can register with the server using a key different 
+from their identity key.
+This will hide their real identity key.
+
+This identity key is effectively a secret and SHOULD only be disclosed to clients that you the user wants to be notified by.
+
+A client MAY advertise the access token on the contact-code topic of the key generated.
+A client MAY share their identity key through [contact updates](./6-payloads.md#contact-update)
+
+A client receiving a push notification identity key SHOULD listen to the contact code 
+topic of the push notification identity key for updates.
+
+If a client has received a push notification identity key they MAY request 
+the access token using an ephemeral key when querying the server.
+
+The method described above effectively does not share the identity of the sender
+nor the receiver to the server, but MAY result in missing push notifications as
+the propagation of the secret is left to the client.
+
+This can be mitigated by [device syncing](./6-payloads.md), but not completely
+addressed.
+
+[//]: This section can be removed, for now leaving it here in order to help with the 
+review process. Point can be integrated, suggestion welcome.
+
+## Security considerations
+
+If no anonymous mode is used, when registering with a push notification service a client discloses:
+
+- Their public key
+- Which devices they want to receive notifications on
+
+They MAY disclose:
```

Comment:
Does "They" refer to a client?

Reply:
yes, changed to avoid possessive adjectives